cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -140,6 +152,7 @@ public void deleteDiskEncryptionSet(DiskEncryptionSetDeletionRequest diskEncrypt
                 }
                 LOGGER.info("Deleting Disk Encryption Set \"{}\"", diskEncryptionSetId);
                 deleteDiskEncryptionSetOnCloud(azureClient, desResourceGroupName, diskEncryptionSetName);
+                checkAndDeleteDesResourceGroupName(azureClient, desResourceGroupName);
```

Comment:
This invocation should take place based on the existence of a `CloudResource` of type `ResourceType.AZURE_RESOURCE_GROUP`. Like done here for the `AZURE_DISK_ENCRYPTION_SET`, the `AZURE_RESOURCE_GROUP` should be looked up in `diskEncryptionSetDeletionRequest.getCloudResources()` in a separate statement outside of this `if` block. `checkAndDeleteDesResourceGroupName()` shall then be called only if a matching `CloudResource` is found for `AZURE_RESOURCE_GROUP`.